src/Our.Umbraco.StorageProviders.AWSS3/IO/AWSS3FileSystem.cs

```diff
@@ -236,18 +256,33 @@ public bool FileExists(string path)
             var request = new GetObjectMetadataRequest
             {
                 BucketName = _bucketName,
-                Key = ResolveBucketPath(path)
+                Key = ResolveBucketPath(path),
             };
 
             try
             {
-                Execute(client => client.GetObjectMetadataAsync(request));
+                var response = Execute(client => client.GetObjectMetadataAsync(request)).Result;
```

Comment:
Using `Result` will wrap any exceptions in an `AggregateException` and in this case, wraps the entire `Execute()` method, causing the try/catch and rethrowing within that method to not work as expected.

By using `GetAwaiter().GetResult()` on the actual async method should fix this:
```suggestion
                var response = Execute(client => client.GetObjectMetadataAsync(request).ConfigureAwait(false).GetAwaiter().GetResult());
```

Reply:
Have updated here as, as you say, it does clean up the exception handling, but will leave any changes to other parts of the code to the decision of the maintainer.